include/boost/math/tools/polynomial.hpp

```diff
@@ -300,6 +307,18 @@ inline polynomial<T> operator * (const U& a, const polynomial<T>& b)
    return result;
 }
 
+template <typename T>
+bool operator == (polynomial<T> const &a, polynomial<T> const &b)
+{
+    return a.data() == b.data();
+}
+
+template <typename T>
+bool operator < (polynomial<T> const &a, polynomial<T> const &b)
+{
+    return a.data() < b.data();
+}
```

Comment:
Hmmm, hold on, I just realized that maybe polynomial order is not as straightforward as lexicographical order. I will sleep on it for now.